src/librustdoc/html/markdown.rs

```diff
@@ -1064,7 +1064,7 @@ pub fn plain_text_summary(md: &str) -> String {
     s
 }
 
-pub fn markdown_links(md: &str) -> Vec<(String, Option<Range<usize>>)> {
+pub fn markdown_links(md: &str) -> Vec<(LinkType, String, Option<Range<usize>>)> {
```

Comment:
Hmm, this is getting large enough it might be worth turning into a `struct MarkdownLink { kind, url, span }`.